irohad/ametsuchi/impl/postgres_block_query.cpp

```diff
@@ -196,30 +198,28 @@ namespace iroha {
     boost::optional<BlockQuery::wTransaction>
     PostgresBlockQuery::getTxByHashSync(
         const shared_model::crypto::Hash &hash) {
-      return getBlockId(hash) |
-          [this](auto blockId) { return block_store_.get(blockId); } |
-          [this](auto bytes) {
-            // TODO IR-975 victordrobny 12.02.2018 convert directly to
-            // shared_model::proto::Block after FlatFile will be reworked to new
-            // model
-            return model::converters::stringToJson(bytesToString(bytes));
-          }
-      | [&](const auto &json) { return serializer_.deserialize(json); } |
-          [](const auto &block) {
-            return boost::optional<shared_model::proto::Block>(
-                shared_model::proto::from_old(block));
-          }
-      | [&](const auto &block) {
+      auto block = getBlockId(hash) |
+          [this](auto block_id) { return block_store_.get(block_id);
+         } | [this](auto bytes) {
```

Comment:
auto defaults to pass-by-value, consider using auto &bytes. Also check other lambdas